common.go

```diff
@@ -204,29 +230,26 @@ func (h *Handler) Handle(b []byte) error {
 // data off.
 func (h *Handler) TransformAndSend(c *Container) error {
 	if err := h.Transform(c); err != nil {
-		return Error{Source: "handler", Reason: "transforming metrics failed", Next: err}
+		return Error{Source: "handler transform&send", Reason: "transforming metrics failed", Next: err}
 	}
-	if err := h.Sender.Send(c); err != nil {
-		return Error{Source: "handler", Reason: "sending metrics failed", Next: err}
+	if err := h.Send(c); err != nil {
+		return Error{Source: "handler transform&send", Reason: "sending metrics failed", Next: err}
 	}
 	return nil
 }
 
 // Verify the basic integrity of a handler. Quite shallow.
 func (h Handler) Verify() error {
-	if h.Parser == nil {
-		return Error{Reason: "Missing parser for Handler"}
-	}
-	if h.Transformers == nil {
-		return Error{Reason: "Missing parser for Handler"}
+	if h.parser == nil {
+		return Error{Source: "handler verification", Reason: "Missing parser for Handler"}
```

Comment:
At first glance, this one was problematic to me, because, e.g., the logrus receiver doesn't _use_ a parser. Then I realized we already had this test, so obviously it worked.... It turns out we default the parser to "json" if one isn't set in configuration. This makes a lot of sense in general. Most of the time, that's what we want, and for stuff like the logrus receiver, it doesn't matter what the parser is.

So this is actually fine. It will catch corrupt or bugged handlers. Which is what we want anyway.